lib/copy-sync/__tests__/copy-sync-case-insensitive-paths.test.js

```diff
@@ -0,0 +1,115 @@
+'use strict'
+
+const assert = require('assert')
+const os = require('os')
+const path = require('path')
+const fs = require(process.cwd())
+
+/* global beforeEach, afterEach, describe, it */
+
+describe('+ copySync() - case insensitive paths', () => {
+  let TEST_DIR = ''
+  let src = ''
+  let dest = ''
+
+  beforeEach(done => {
+    TEST_DIR = path.join(os.tmpdir(), 'fs-extra', 'copy-sync-case-insensitive-paths')
+    fs.emptyDir(TEST_DIR, done)
+  })
+
+  afterEach(done => fs.remove(TEST_DIR, done))
+
+  describe('> when the source is a directory', () => {
+    it('should behave correctly based on the OS', () => {
+      src = path.join(TEST_DIR, 'srcdir')
+      fs.outputFileSync(path.join(src, 'subdir', 'file.txt'), 'some data')
+      dest = path.join(TEST_DIR, 'srcDir')
+
+      try {
+        fs.copySync(src, dest)
```

Comment:
What if this function won't throw. The test will pass, as the assertions won't get called, right? Shouldn't we throw any error right after `fs.copySync` so we make sure we trigger the `catch? 

```js
try {
  fs.copySync(src, dest)
  throw new Error('fs.copySync didn\'t throw')
} ...

Reply:
@JPeer264 that's a good point! however, we specifically want to test if `copySync` throws that error. So, I wouldn't throw error manually there, but I think the better way to make sure `copySync` throws that error would be to assert a boolean after try-catch. I'll apply that there.